header/p2p/peer_tracker.go

```diff
@@ -37,15 +63,18 @@ func (p *peerTracker) track(ctx context.Context) {
 	subs, err := p.host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
 	if err != nil {
 		log.Errorw("subscribing to EvtPeerConnectednessChanged", "err", err)
+		p.done <- struct{}{}
```

Comment:
defer this at the top of the func so you don't have to call at every return.

Reply:
yes, thanks